addon/components/react-component.js

```diff
@@ -38,9 +38,26 @@ const ReactComponent = Ember.Component.extend({
 
     const props = getMutableAttributes(get(this, 'attrs'));
 
+    // Determine the children
+    // If there is already `children` in `props`, we just pass it down (it can be function).
+    // Otherwise we need to wrap the current `childNodes` inside a React component.
+    // It is important that `childNodes` are reconstructed with `[...childNodes]` because
+    // it is a `NodeList`-type object instead of Array in the first place.
+    // Without reconstructing, `childNodes` will include the React component itself when
+    // `componentDidMount` hook is triggerred.
+    let children = props.children;
+    if (!children) {
+      const childNodes = get(this, 'element.childNodes');
+      children = [React.createElement(EmberWrapper, {
+        key: get(this, 'elementId'),
```

Comment:
Since `children` here is always a single element array I think you can just set `key: '0'`